File: python/invoice_item.py
class InvoiceItem:
  def __init__(self, spend_type, amount, transaction):
    self.spend_type = spend_type
    self.amount = amount
    self.transaction = transaction

  def __str__(self):
    return f"InvoiceItem for ${self.amount}: {self.spend_type} for {self.transaction.name}"

File: ruby/invoice_item.rb
class InvoiceItem
  attr_reader :spend_type, :amount, :transaction

  def initialize(spend_type:, amount:, transaction:)
    @spend_type = spend_type
    @amount = amount
    @transaction = transaction
  end

  def to_s
    "InvoiceItem for $#{@amount}: #{@spend_type} for #{@transaction.name}"
  end
end
